### PR TITLE
[JBPM-9178] Immutable springboot images / adding the capability to re…

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9178

### CODE REVIEW

**Review by afalhambra — CHANGES_REQUESTED** [review]
Looks good, nothing major, cosmetic changes.

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java [hunk 1]
```diff
@@ -66,18 +81,23 @@ protected static DeploymentDescriptor getDescriptorFromKModule(InternalKieModule
...
+        ByteArrayInputStream input = new ByteArrayInputStream(content);
+        try {
+            return DeploymentDescriptorIO.fromXml(input);
+        } finally {
+            try {
+                input.close();
+            } catch (IOException e) {
+                logger.debug("Error when closing stream of kie-deployment-descriptor.xml");
+            }
+        }
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java:101]: done

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java [hunk 2]
```diff
@@ -66,18 +81,23 @@ protected static DeploymentDescriptor getDescriptorFromKModule(InternalKieModule
...
+
+    protected static DeploymentDescriptor getDescriptorFromContent(byte[] content) {
+        ByteArrayInputStream input = new ByteArrayInputStream(content);
+        try {
+            return DeploymentDescriptorIO.fromXml(input);
+        } finally {
+            try {
+                input.close();
+            } catch (IOException e) {
+                logger.debug("Error when closing stream of kie-deployment-descriptor.xml");
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java:99]: ok

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java [hunk 3]
```diff
@@ -38,9 +42,20 @@
...
+        if (kieProject instanceof KieModuleKieProject) {
+            InternalKieModule module = ((KieModuleKieProject) kieProject).getInternalKieModule();
             collectDeploymentDescriptors(module, descriptorHierarchy);
+        } else if (kieProject instanceof ClasspathKieProject) {
+            InputStream is = ((ClasspathKieProject) kieProject).getClassLoader().getResourceAsStream(DeploymentDescriptor.META_INF_LOCATION);
+            if(is != null) {
+                try {
+                    descriptorHierarchy.add(getDescriptorFromContent(IoUtils.readBytesFromInputStream(is)));
+                } catch (IOException e) {
+                    logger.warn("Cound not read deployment descriptor on classpath", e);
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java:55]: ok

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (afalhambra) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java]: I can't see this yet?

**Review by afalhambra — CHANGES_REQUESTED** [review]
Sorry, can't see the change you marked as "resolved" to add more details about the IOException in the warning log

https://github.com/kiegroup/jbpm/blob/d090fe4588afd8b585405e4e9d96b51dcf2d296e/jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java#L95

**Comment by elguardian** [comment]
> Sorry, can't see the change you marked as "resolved" to add more details about the IOException in the warning log
> 
> https://github.com/kiegroup/jbpm/blob/d090fe4588afd8b585405e4e9d96b51dcf2d296e/jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/deploy/DeploymentDescriptorManagerUtil.java#L95

Aside to set is as warn there is no point as that is in memory descriptor already. so the only possibility is a failure in the parsing.

**Review by afalhambra — APPROVED** [review]
Great work, really hope this is the last change regarding immutable springboot :)

**Pull request merged**